### DISCUSSION TITLE
Stack too deep

### DISCUSSION THREAD

**Discussion by JDEEPSingh04** [Cyfrin/foundry-full-course-cu#1036] [Q&A]
Hey!! I was working on the Raffle contract. When I passed the private Key in the vm.startBroadcast, I got an unexpected error 
Error: Compiler error (/solidity/libsolidity/codegen/CompilerUtils.cpp:1434):Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.

Please help me with this error.

I have tried using --via-ir but it's compliling but takes a lot of time for compilation (nearly 1 min)
And still when i run forge coverage with a fork url I get the same error again

Kindly help me in resolving this error

**Comment by theirrationalone** [comment]
@usmanfarooq91 ! To the rescue.... ✋✋;`)

**Reply by 0xusmanf** [reply]
Sorry, I couldn't respond earlier as my mother had knee surgery and I was busy there but I am back now. I hope you have solved this issue by now but if you still didn't, [this](https://github.com/Cyfrin/foundry-full-course-f23/discussions/851#discussioncomment-6992652) might help.

**Reply by theirrationalone** [reply]
How's your mother right now???? I wish she is all good, healthy and blessed ❤❤❤

**Reply by theirrationalone** [reply]
@usmanfarooq91 ! Please take care of Her first. ❤❤

**Reply by 0xusmanf** [reply]
Thank you for your well wishes ❤️ , she is healthy and recovering fast.

**Reply by alfheimrShiven** [reply]
Good to hear that she's recovering @usmanfarooq91 . Take care.

**Comment by alfheimrShiven** [comment]
Hey @JDEEPSingh04 the standard `Stack too deep` error is thrown by the Solidity compiler when:
- the contract(s) use a lot of local variables
- have complex logical expressions
- have a deep function stack

Since in the Lottery project, a ton of local variables are used to configure the `Raffle.sol` contract, that's more likely to be causing this. It would be helpful if you can share the project repo here, as it would require some debugging to point out the exact file causing this.

**Comment by Ezejaemmanuel** [comment]
i am having the problem also

**Reply by alfheimrShiven** [reply]
@Ezejaemmanuel kindly share your repo
